### PR TITLE
Base installation flow in operator

### PR DESCRIPTION
**Pull request by mishoyama**
### Issue dell/csi-baremetal#316

CSI deployment with a set of limitations:
- All values hardcoded
- No node operator deployment
- No CRD deployment
- No RBAC/SC/ConfigMap deployment

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [x] All TODOs are linked with the issues
- [x] All comments are resolved

## Testing
1. Deploy
```
borism1@ubuntu:/workspace/csi-baremetal-operator$ make run
go fmt ./...
go vet ./...
go run ./main.go
2021-03-18T15:49:40.624Z        INFO    controller-runtime.metrics      metrics server is starting to listen    {"addr": ":8080"}
2021-03-18T15:49:40.624Z        INFO    setup   starting manager
2021-03-18T15:49:40.625Z        INFO    controller-runtime.manager      starting metrics server {"path": "/metrics"}
2021-03-18T15:49:40.625Z        INFO    controller-runtime.controller   Starting EventSource    {"controller": "deployment", "source": "kind source: /, Kind="}
2021-03-18T15:49:40.726Z        INFO    controller-runtime.controller   Starting Controller     {"controller": "deployment"}
2021-03-18T15:49:40.726Z        INFO    controller-runtime.controller   Starting workers        {"controller": "deployment", "worker count": 1}
2021-03-18T15:49:40.726Z        INFO    controllers.Deployment  Custom resource obtained        {"deployment": "/csi"}
2021-03-18T15:49:40.743Z        INFO    Daemon set created successfully {"csi-baremetal": "node"}
2021-03-18T15:49:40.773Z        INFO    Deployment created successfully {"csi-baremetal": "controller"}
2021-03-18T15:49:40.813Z        INFO    Daemon set created successfully {"csi-baremetal": "extender"}
2021-03-18T15:49:40.886Z        INFO    Daemon set created successfully {"csi-baremetal": "patcher"}
2021-03-18T15:49:40.886Z        DEBUG   controller-runtime.controller   Successfully Reconciled {"controller": "deployment", "request": "/csi"}
```
2. Verify pods:
```
borism1@ubuntu:/workspace/csi-baremetal-operator$ pods
NAME                                        READY   STATUS    RESTARTS   AGE   IP            NODE                 NOMINATED NODE   READINESS GATES
csi-baremetal-controller-6fd7f58987-hxbsd   4/4     Running   0          58s   10.244.5.23   kind-worker4         <none>           <none>
csi-baremetal-node-7vcf6                    4/4     Running   0          58s   10.244.1.22   kind-worker3         <none>           <none>
csi-baremetal-node-h9vk6                    4/4     Running   0          58s   10.244.5.22   kind-worker4         <none>           <none>
csi-baremetal-node-k82wc                    4/4     Running   0          58s   10.244.3.24   kind-worker6         <none>           <none>
csi-baremetal-node-qv5lv                    4/4     Running   0          58s   10.244.6.22   kind-worker5         <none>           <none>
csi-baremetal-node-vhgb2                    4/4     Running   0          58s   10.244.2.20   kind-worker          <none>           <none>
csi-baremetal-node-zvx4x                    4/4     Running   0          58s   10.244.4.20   kind-worker2         <none>           <none>
csi-baremetal-se-kqwzj                      1/1     Running   0          58s   172.18.0.3    kind-control-plane   <none>           <none>
csi-baremetal-se-patcher-mlvn7              1/1     Running   0          58s   10.244.0.14   kind-control-plane   <none>           <none>
```

### CODE REVIEW

File: pkg/common.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	_, err := dsClient.Get(name, metav1.GetOptions{})
+	return isFound(err)
+}
+
+func isDeploymentDeployed(dsClient appsv1.DeploymentInterface, name string) (bool, error) {
+	_, err := dsClient.Get(name, metav1.GetOptions{})
+	return isFound(err)
+}
+
+func isFound(err error) (bool, error) {
```
Comment (AnastasiaShemyakinskaya) [inline pkg/common.go:101]: I think, kubernetes lib has similar function: https://github.com/dell/csi-baremetal/blob/008bda22eb89a317ea6b7e4b0e5da2521b8cfaed/pkg/base/k8s/kubeclient.go#L287

File: main.go [hunk 2]
```diff
@@ -21,13 +21,15 @@ import (
 	"os"
 
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
```
Comment (AnastasiaShemyakinskaya) [inline main.go:24]: Controller-runtime has kubernetes client, why to use client from client-go?

File: pkg/common.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
+package pkg
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (AnastasiaShemyakinskaya) [inline pkg/common.go:6]: Better to keep import order and format files with gofmt

File: main.go (hunk 2 above)
Comment (mishoyama) [reply main.go:24]: `client-go` simplifies deployment of StatefulSet, ReplicaSet, etc.

File: pkg/common.go (hunk 3 above)
Comment (mishoyama) [reply pkg/common.go:6]: Updated.

File: pkg/common.go (hunk 1 above)
Comment (mishoyama) [reply pkg/common.go:99]: ok, thanks! refactored

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

File: pkg/node.go [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: loopbackManagerConfigName},
+				Optional:             pointer.BoolPtr(true),
+			},
+		}},
+	}
+}
+
+func createNodeContainers() []corev1.Container {
+	// todo get rid of this hack
```
Comment (painhardcore) [inline pkg/node.go:163]: link TODO with the issue?

**Review by painhardcore — CHANGES_REQUESTED** [review]

**Review by safronovD — APPROVED** [review]

File: pkg/node.go (hunk 4 above)
Comment (mishoyama) [reply pkg/node.go]: Done. And removed extra TODOs which already planned in issues.

**Review by painhardcore — APPROVED** [review]

**Pull request merged**